src/main/java/org/team4159/frc/robot/commands/auto/FollowTrajectory.java

```diff
@@ -0,0 +1,136 @@
+package org.team4159.frc.robot.commands.auto;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.controller.RamseteController;
+import edu.wpi.first.wpilibj.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Transform2d;
+import edu.wpi.first.wpilibj.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.trajectory.Trajectory;
+import edu.wpi.first.wpilibj2.command.*;
+
+import org.team4159.frc.robot.subsystems.Drivetrain;
+
+import static org.team4159.frc.robot.Constants.*;
+
+public class FollowTrajectory extends CommandBase {
+  private Trajectory traj;
+  private Drivetrain drivetrain;
+
+  private Timer timer = new Timer();
+
+  // private CSVWriter writer;
+
+  private DifferentialDriveWheelSpeeds prev_speeds = new DifferentialDriveWheelSpeeds(0,0);
+  private DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(DRIVE_CONSTANTS.TRACK_WIDTH);
+  private SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(DRIVE_CONSTANTS.kS, DRIVE_CONSTANTS.kV, DRIVE_CONSTANTS.kA);
+  private RamseteController controller = new RamseteController(DRIVE_CONSTANTS.kB, DRIVE_CONSTANTS.kZeta);
+  private PIDController pid_left = new PIDController(DRIVE_CONSTANTS.kP, 0, DRIVE_CONSTANTS.kD);
+  private PIDController pid_right = new PIDController(DRIVE_CONSTANTS.kP, 0, DRIVE_CONSTANTS.kD);
+  double prev_time = 0;
+
+  public FollowTrajectory(Trajectory trajectory, Drivetrain drivetrain) {
+    this.drivetrain = drivetrain;
+
+    Transform2d transform = drivetrain.getPose().minus(trajectory.getInitialPose());
+    traj = trajectory.transformBy(transform);
+
+    addRequirements(drivetrain);
+  }
+
+  @Override
+  public void initialize() {
+    // writer = new CSVWriter("/home/lvuser/Output");
+
+    prev_time = 0;
+    var initialState = traj.sample(0);
+    prev_speeds = kinematics.toWheelSpeeds(
+      new ChassisSpeeds(initialState.velocityMetersPerSecond,
+        0,
+        initialState.curvatureRadPerMeter
+          * initialState.velocityMetersPerSecond));
+    timer.reset();
+    timer.start();
+  }
+
+  @Override
+  public void execute() {
```

Comment:
maybe just migrate to RamseteCommand? (unless we want to keep for smart dashboard stuff)

Reply:
going to add logging stuff eventually